timeseries/src/autogluon/timeseries/utils/features.py

```diff
@@ -209,9 +227,47 @@ def fit(self, data: TimeSeriesDataFrame) -> None:
             static_features_cat=static_features_cat,
             static_features_real=static_features_real,
         )
-        self._train_covariates_real_median = data[self.covariate_metadata.covariates_real].median()
+
+        # Median of real-valued covariates will be used for missing value imputation
+        if self.num_samples is not None and len(df) > self.num_samples:
+            df = df.sample(n=self.num_samples, replace=True)
+        self._train_covariates_real_median = df[self.covariate_metadata.covariates_real].median()
+
+        self.fit_time = time.monotonic() - start_time
         self._is_fit = True
 
+        df_out = self._concat_dfs(dfs_to_concat)
+        df_out.index = index
```

Comment:
Are we sure that no prior ops change the index? Doesn't look like it.

Reply:
Do you mean that `df_out` has the same rows / in the same order as the original `df`? I am almost certain that this is the case since none of the steps in `ContinuousAndCategoricalFeatureGenerator` change the order of the rows (if they did, this would also break AutoGluon-Tabular). 

If you rather mean "are we sure that the `index` object (defined on [L163](https://github.com/autogluon/autogluon/pull/4557/files#diff-ac94905e2a5e94bcd2c6b86f47354f212aa012314125dbad34c9d196fa432b12R163)) is not modified by any transformations" - I've just added an extra assertion to the [test](https://github.com/autogluon/autogluon/pull/4557/files#diff-a634dd5750d533af15c67f8e7cb99046d5156bf43c70de32216a8ff4136dbb5cR123) to ensure that the index object is not modified.

Reply:
I meant the former. Thanks for adding the test.